packages/x-components/src/x-modules/facets-next/__tests__/facets-service.spec.ts

```diff
@@ -0,0 +1,496 @@
+        )
+      ).toBe(false);
+      expect(
+        service.areFiltersDifferent([createRawFilter('size:l')], [createRawFilter('size:l')])
+      ).toBe(false);
+      expect(
+        service.areFiltersDifferent(
+          [
+            createNextSimpleFilter('color', 'red', false),
+            createNextHierarchicalFilter('category', 'shirt'),
+            createNextNumberRangeFilter('price', { min: null, max: 10 }),
+            createNextEditableNumberRangeFilter('age', { min: null, max: 30 }),
+            createRawFilter('size:l')
+          ],
+          [
+            createRawFilter('size:l'),
+            createNextEditableNumberRangeFilter('age', { min: null, max: 30 }),
+            createNextNumberRangeFilter('price', { min: null, max: 10 }),
+            createNextHierarchicalFilter('category', 'shirt'),
+            createNextSimpleFilter('color', 'red', false)
+          ]
+        )
+      ).toBe(false);
+    });
+  });
+
+  describe('clears filters', () => {
+    it('deselects all filters', () => {
+      const { service, getSelectedFilters } = prepareFacetsService();
+
+      [
+        createRawFilter('size:l'),
+        createNextEditableNumberRangeFilter('category', { min: null, max: 30 }),
+        createNextNumberRangeFilter('category', { min: null, max: 10 }),
+        createNextHierarchicalFilter('category', 'shirt'),
+        createNextSimpleFilter('color', 'red', false)
+      ].forEach(filter => {
+        service.select(filter);
+      });
+      expect(getSelectedFilters()).toHaveLength(5);
+
+      service.clearFilters();
+      expect(getSelectedFilters()).toHaveLength(0);
+    });
+
+    it('deselects only filters from the given facets ids', () => {
+      const { service, getSelectedFilters, isFilterSelected, isEditableNumberRangeFilterSelected } =
+        prepareFacetsService();
+
+      const filters = [
+        createRawFilter('size:l'),
+        createNextEditableNumberRangeFilter('age', { min: null, max: 30 }),
+        createNextNumberRangeFilter('price', { min: null, max: 10 }),
+        createNextHierarchicalFilter('category', 'shirt'),
+        createNextSimpleFilter('color', 'red', false)
+      ];
+      filters.forEach(filter => {
+        service.select(filter);
+      });
+      expect(getSelectedFilters()).toHaveLength(5);
+
+      const [sizeFilter, ageFilter, priceFilter] = filters;
+      service.clearFilters(['category', 'color']);
+      expect(getSelectedFilters()).toHaveLength(3);
+      expect(isFilterSelected(sizeFilter)).toBe(true);
+      expect(isFilterSelected(priceFilter)).toBe(true);
+      expect(isEditableNumberRangeFilterSelected(ageFilter as EditableNumberRangeFilter)).toBe(
+        true
+      );
+    });
+  });
+  describe('saves a group of facets', () => {
+    it('saves groups of facets keeping its previous selected states', () => {
+      const { service, getSelectedFilters, getFilters, getStoreEditableNumberRangeFilter } =
```

Comment:
I'd like to have some time to rethink this test. It represents a real use case, but it is too verbose.

Reply:
It's ok as it is. Long and verbose? yes, does it test correctly? yes

let's move on!